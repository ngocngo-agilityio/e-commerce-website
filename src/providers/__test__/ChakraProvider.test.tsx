// Libs
import { render, screen } from '@testing-library/react';

// Components
import { ChakraUIProvider } from '..';

jest.mock('next/font/google', () => ({
  Lato: jest.fn().mockReturnValue({ style: { fontFamily: 'lato' } }),
  Arimo: jest.fn().mockReturnValue({ style: { fontFamily: 'arimo' } }),
  Baloo_2: jest.fn().mockReturnValue({ style: { fontFamily: 'baloo' } }),
  Inter: jest.fn().mockReturnValue({ style: { fontFamily: 'inter' } }),
}));

describe('ChakraUIProvider component', () => {
  test('should render ChakraUIProvider successfully', () => {
    render(
      <ChakraUIProvider>
        <div>Test</div>
      </ChakraUIProvider>,
    );

    expect(screen.getByText('Test')).toBeInTheDocument();
  });

  test('should match snapshot for ChakraUIProvider', () => {
    const { container } = render(
      <ChakraUIProvider>
        <div>Test</div>
      </ChakraUIProvider>,
    );

    expect(container).toMatchSnapshot();
  });
});
