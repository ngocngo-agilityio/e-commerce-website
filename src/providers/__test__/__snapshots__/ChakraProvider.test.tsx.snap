// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChakraUIProvider component should match snapshot for ChakraUIProvider 1`] = `
<div>
  <div>
    Test
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
